src/vm.c

```diff
@@ -0,0 +1,22 @@
+#include "vm.h"
+
+void init_vm(pi_vm_t* vm, int size, int* err) {
+
+	if ( (size < VM_SIZE_MIN) || (size > VM_SIZE_MAX) ) {
+		*err = VM_ERR_INVALID_SIZE;
+		return;
+	}
+
+	vm->cells = malloc( size * sizeof(vm_cell_t) );
+	if (vm->cells == NULL) {
+		*err = VM_ERR_INVALID_ALLOC;
+		return;
+	}
+	vm->cell_ct = size;
+
+	/* Set everything to 0 to begin */
+	memset(vm->cells, 0, size);
```

Comment:
Does this set everything to zero? if this is memory, can't we just allocate without needing to set it to a value?

Reply:
Yeah, but I don't want the VM to segfault if it tries to reference an uninitialized value.

Reply:
true, we should later add something to know if the location in mem is used, idk sounds good but maybe more inefficient. Looks great!